pkg/transport/http/message_write_responsewriter.go

```diff
@@ -15,15 +16,24 @@ import (
 
 // Write out to the the provided httpResponseWriter with the message m.
 // Using context you can tweak the encoding processing (more details on binding.Write documentation).
-func WriteResponseWriter(ctx context.Context, m binding.Message, rw http.ResponseWriter, transformers binding.TransformerFactories) error {
-	structuredWriter := &httpResponseWriterEncoder{rw: rw}
-	binaryWriter := &httpResponseWriterEncoder{rw: rw}
+func WriteResponseWriter(ctx context.Context, m binding.Message, er event.Result, rw http.ResponseWriter, transformers binding.TransformerFactories) error {
+	status := http.StatusOK
+	if er != nil {
```

Comment:
Are you sure this logic should live here? I see this logic more "client related": IMO this function should take `statusCode` as parameter, and the caller performs the transformation `event.Result` to status code

Reply:
the way the `ResponseWriter` works is if you call `rw.Write(body)` and no headers have been set, it finalizes the headers with the default `http.StatusOK`. So unlike the other messages in other protocols, the ResponseWriter is an active write out the wire. There is no chance to call a `.Send` or `.Do` style method.